components/places/src/storage/bookmarks/conversions.rs

```diff
@@ -0,0 +1,288 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use super::{
+    BookmarkPosition, BookmarkRootGuid, BookmarkTreeNode, InsertableBookmark, InsertableFolder,
+    InsertableItem, InsertableSeparator, PublicNode, RawBookmark, UpdatableBookmark,
+    UpdatableFolder, UpdatableItem, UpdatableSeparator, UpdateTreeLocation,
+};
+
+use crate::error::{InvalidPlaceInfo, Result};
+use crate::msg_types;
+use crate::types::{BookmarkType, SyncGuid};
+use url::Url;
+
+impl From<BookmarkTreeNode> for PublicNode {
```

Comment:
I think the public API could live in `places::api` (at least, that's what I understood `api` is for, but maybe that's changed), but that's a small nit.

Reply:
It's really "operations on PublicNode" more than "the publically facing portion of the API". I'm not sure how to better split things up, to be honest. I'll try renaming this to public_node, maybe.